lib/nanoc/extra/link_collector.rb

```diff
@@ -4,6 +4,18 @@
 
 module ::Nanoc::Extra
   class LinkCollector
+
+    @@uri_attrs = {
+      'a' => :href,
```

Comment:
1. Prevent insertion of insecure "active content" (such as JavaScript) that could generate misleading content or steal private information
2. Prevent passive leaking of information--HTTP requests are sent in-the-clear and might leak information about the user's browsing session

Users who navigate using insecure hyperlinks are susceptible to `#2`, but navigation is performed actively by the user, and it's harder to achieve the task of "saving users from themselves". Plus, the current site's maintainer has no control over the availability of SSL on remote servers.